src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,238 @@
+        iDynTree::Vector2 m_currentDcmPosition; /**< The current DCM position.*/
+
+        double m_remainingSingleSupportDuration;/**< The remained single support duration.*/
+        double m_stepTiming; /**< The remanined single support duration+(next double support duration)/2  that is used for optimization.*/
+        double m_stepDurationTolerance;/**< The tolerance of step timing with respect to the nominal value.*/
+
+        double m_omega;/**< The natural frequency of LIPM.*/
+
+        double m_currentTime;/**< The  current time.*/
+        double m_nextDoubleSupportDuration;/**< The timing of next double support.*/
+
+        /**
+         *The buffered vectors for the interpolation of the foot trajectory
+         */
+        iDynTree::VectorDynSize m_xPositionsBuffer, m_yPositionsBuffer, m_zFirstPiecePositionsBuffer,m_zSecondPiecePositionsBuffer,
+                                m_yawsBuffer, m_timesBuffer, m_zFirstPieceTimesBuffer,m_zSecondPieceTimesBuffer;
+
+        iDynTree::ConvexHullProjectionConstraint m_convexHullComputer; /**< iDynTree convex hull helper. */
+        std::vector<iDynTree::Polygon> m_feetExtendedPolygon;/**< convex hull of the allowable landing foot position. */
+        iDynTree::Transform m_footTransform; /**< transform of the next foot position. */
+
+        bool m_isSolutionEvaluated{false}; /**< True if the solution is evaluated. */
+
+        /**
+         * Compute the hessian matrix.
+         * Please do not call this function to update the hessian matrix! It can be set only once.
+         * @return true/false in case of success/failure.
+         */
+        bool computeHessianMatrix();
+
+        /**
+         * Compute or update the linear constraints matrix(A) related to equality and inequality constraints(C<Ax<B)
+         * If the solver is already set the linear constraints matrix is updated otherwise it is set for
+         * the first time.
+         * @return true/false in case of success/failure.
+         */
+        bool computeConstraintsMatrix();
+
+        /**
+         * Compute or update the gradient
+         * @return true/false in case of success/failure.
+         */
+        bool computeGradientVector();
+
+        /**
+         * Set or update the lower and the upper bounds
+         * @return true/false in case of success/failure.
+         */
+        bool computeBoundsVectorOfConstraints();
+
+    public:
+        /**
+         * Constructor of step adaptation controller.
+         * @param inputSize size of the controlled input vector;
+         * @param numberOfAllConstraints number of equality and inequality constraints!
+         */
+        StepAdaptationController();
+
+        /**
+         * Initialize the method
+         * @param config yarp searchable configuration variable.
+         * @return true/false in case of success/failure
+         */
+        bool Configure(const yarp::os::Searchable& config);
```

Comment:
For class methods, please use a lowercase first letter. Uppercase first letters are for class names

Reply:
Done!